Mage.Sets/src/mage/cards/d/DesertWarfare.java

```diff
@@ -0,0 +1,225 @@
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.ZoneChangeEvent;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.SandWarriorToken;
+import mage.game.permanent.token.Token;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+import mage.target.targetpointer.FixedTargets;
+
+import java.util.UUID;
+
+/**
+ * @author Sidorovich77
+ */
+public final class DesertWarfare extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterPermanent();
+
+    static {
+        filter.add(SubType.DESERT.getPredicate());
+    }
+
+    public DesertWarfare(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{3}{G}");
+
+        // Whenever you sacrifice a Desert and whenever a Desert card is put into your graveyard from your hand or library, put that card onto the battlefield under your control at the beginning of your next end step.
+        //Based on Seraph, Yuma, Proud Protector
+        Effect effect = new CreateDelayedTriggeredAbilityEffect(new AtTheBeginOfNextEndStepDelayedTriggeredAbility(new DesertWarfareReturnEffect(), TargetController.YOU));
+        effect.setText("put that card onto the battlefield under your control at the beginning of your next end step");
+
+        this.addAbility(new DesertWarfareTriggeredAbility(Zone.BATTLEFIELD, effect, false, TargetController.YOU, SetTargetPointer.CARD));
+
+        //At the beginning of combat on your turn, if you control five or more Deserts, create that many 1/1 red, green, and white Sand Warrior creature tokens. They gain haste.
+        //Based on Palani's Hatcher
+        this.addAbility(new ConditionalInterveningIfTriggeredAbility(
+                new BeginningOfCombatTriggeredAbility(new DesertWarfareCreateTokensEffect(), TargetController.YOU, false),
+                new PermanentsOnTheBattlefieldCondition(filter, ComparisonType.OR_GREATER, 5, true),
+                "At the beginning of combat on your turn, if you control five or more Deserts, " +
+                        "create that many 1/1 red, green, and white Sand Warrior creature tokens. They gain haste.")
+                .addHint(DesertsYouControlCount.getHint()));
+    }
+
+    private DesertWarfare(final DesertWarfare card) {
+        super(card);
+    }
+
+    @Override
+    public DesertWarfare copy() {
+        return new DesertWarfare(this);
+    }
+
+}
+
+//Based on PutCardIntoGraveFromAnywhereAllTriggeredAbility
+class DesertWarfareTriggeredAbility extends TriggeredAbilityImpl {
+
+    private static FilterOwnedCard filterCard = new FilterOwnedCard();
+
+    static {
+        filterCard.add(SubType.DESERT.getPredicate());
+    }
+
+    private final SetTargetPointer setTargetPointer;
+
+    public DesertWarfareTriggeredAbility(Zone zone, Effect effect, boolean optional, TargetController targetController, SetTargetPointer setTargetPointer) {
```

Comment:
this is a custom class so hardcode all the params and remove the ones you don't need